lib/gitlab/client/events.rb

```diff
@@ -0,0 +1,58 @@
+class Gitlab::Client
+  # Defines methods related to events.
+  # @see https://docs.gitlab.com/ce/api/events.html
+  module Events
+    # Gets a list of authenticated user's events
+    #
+    # @example
+    #   Gitlab.events()
+    #   Gitlab.events({ action: 'created', target_type: 'issue' })
+    #
+    # @param  [Hash] options A customizable set of options.
+    # @option options [String] :action Only events of specific action type
+    # @option options [String] :target_type Only events of specific target type
+    # @option options [Date] :before Only events created before date
```

Comment:
Only thing I think of is are these dates going to be formatted correctly in the query params when in production?

I'll check to see if dates are used elsewhere

Reply:
Example: https://github.com/NARKOZ/gitlab/blob/80960fab45dbf6c9c2b902bc930ab4f8d4d7ae4e/lib/gitlab/client/snippets.rb#L58